pennylane/devices/preprocess.py

```diff
@@ -230,6 +230,7 @@ def warn_about_trainable_observables(
 
 
 @transform
+@qml.QueuingManager.stop_recording()
```

Comment:
What we did until now is that we use `with qml.QueuingManager.stop_recording():` when necessary. Either I would continue doing that here it would become:

```
with qml.QueuingManager.stop_recording():
    new_mp = qml.measurements.StateMP(wires=tape.wires)
```
otherwise I would go with the global solution in the transform dispatcher. I don't like this intermediate solution too much. What do you think?

Reply:
I'd actually prefer to go with a global `TrasnformDipsatcher` solution in the transform dispatcher. It would make transforms easier to write and maintain.